src/db/utils.js

```diff
@@ -0,0 +1,53 @@
+const createCustomDbError = message => {
+  const err = new Error(message)
+  err.name = 'CustomDbError'
+  return err
+}
+
+// static method for Mongoose schema
+const findByIdOrError = function (id) {
+  return new Promise((resolve, reject) => {
```

Comment:
findById seems to return a promise... The promise wrapper is unessisary

```js
findByIdOrError = id => this.findById(id)
  .then(user => {
    if (!user) {
      return Promise.reject(createCustomDbError('No document matching that id'))
    }
    return Promise.resolve(user)
})
```

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise